lib/src/constants/typedefs.dart

```diff
@@ -29,6 +29,7 @@ typedef SpecialItemBuilder<Path> = Widget? Function(
   BuildContext context,
   Path? path,
   int length,
+  bool isPermissionLimited,
```

Comment:
Is this necessary? Would it be better if we provided the builder delegate directly?

Reply:
Yes, it is useful for cases where special item is required to be removed when isPermissionLimited is false. Example as below.

```dart
SpecialItem(
                itemPosition: SpecialItemPosition.append,
                itemBuilder: (
                  BuildContext context,
                  AssetPathEntity? path,
                  int length,
                  bool isPermissionLimited,
                ) {
                  if (!isPermissionLimited) {
                    return null;
                  }
                  return const Center(
                    child: Text('Append Widget', textAlign: TextAlign.center),
                  );
                },
              ),
```

Reply:
Could we use the permission state directly rather than the boolean flag?

Reply:
can, updated isPermissionLimited to permissionState.